controllers/openstackmachine_controller.go

```diff
@@ -422,7 +422,7 @@ func (r *OpenStackMachineReconciler) reconcileMachineState(scope *scope.WithLogg
 		conditions.MarkTrue(openStackMachine, infrav1.InstanceReadyCondition)
 
 		// Set properties required by CAPI machine controller
-		openStackMachine.Spec.ProviderID = ptr.To(fmt.Sprintf("openstack:///%s", *openStackServer.Status.InstanceID))
+		openStackMachine.Spec.ProviderID = ptr.To(fmt.Sprintf("openstack://%s/%s", openStackMachine.Spec.IdentityRef.Region, *openStackServer.Status.InstanceID))
```

Comment:
How can we get some test coverage of this?

Reply:
Whoops! IdentityRef is a pointer and is usually nil.

Incidentally, wouldn't these errors be so much simpler to see if you'd had to write something like `openStackMachine.Spec.IdentityRef->Region` instead? Go automatic dereferencing is insidious.